### PR TITLE
Feature/magicdraw 1127

### PR DESCRIPTION
**Pull request by DavidWillard**

### CODE REVIEW

**Comment by ivan-gomes** [comment]
@DavidWillard ping me when this is ready to merge please.

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,74 +1,115 @@
...
+            no_output_timeout: 1h
+        
+        - save_cache:
+            paths:
+                - ~/.gradle
+            key: gradle-repo-v5-{{ .Branch }}-{{ checksum "gradle.properties" }}
+        
+        - run:
+            name: Compile
+            command: xvfb-run ./gradlew -PbuildNumber=$CIRCLE_BUILD_NUM -PbuildAccess=opensource -PbuildTag=$CIRCLE_TAG --refresh-dependencies --info --stacktrace clean assemble
```
Comment (ivan-gomes) [inline .circleci/config.yml]: Compiling shouldn't require a desktop environment. Let's confirm that and remove `xvfb-run`.

File: .circleci/config.yml [hunk 2]
```diff
@@ -1,74 +1,115 @@
...
+        # Test
+        - run:
+            name: Run tests
+            command: xvfb-run ./gradlew -PbuildAccess=opensource -PbuildNumber=$CIRCLE_BUILD_NUM -PbuildTag=$CIRCLE_TAG --info --stacktrace test
+        
+        - run:
+            name: Collect test results
+            command: find . -type f -regex ".*/build/test-results/.*xml" -exec cp -v {} ~/circleci-test-results/junit/ \;
+        - run:
+            name: Collect aritfacts
```
Comment (ivan-gomes) [inline .circleci/config.yml]: Typo

File: .circleci/config.yml [hunk 3]
```diff
@@ -1,74 +1,115 @@
...
+          # Save test results
+        - store_test_results:
+            path: ~/circleci-test-results
+          # Save artifacts
+        - store_artifacts:
+            path: ~/circleci-artifacts
+        - store_artifacts:
+            path: ~/circleci-test-results
+        
+        - run: rm -rf .git
```
Comment (ivan-gomes) [inline .circleci/config.yml]: Why was removing the `.git` directory necessary? Just curious.

Comment (DavidWillard) [reply .circleci/config.yml]: Just to reduce the size of the workspace before running the persist to workspace command.

Comment (ivan-gomes) [reply .circleci/config.yml]: Doesn't the `paths` property already exclude other directories from being persisted, including `.git`?

Comment (ivan-gomes) [reply .circleci/config.yml]: `.git` is inside `project` folder, so it isn't excluded by default.

**Pull request merged**